package/MDAnalysis/lib/c_distances.pyx

```diff
@@ -267,3 +271,112 @@ def contact_matrix_pbc(coord, sparse_contacts, box, cutoff):
             if dist < cutoff2:
                 sparse_contacts[i, j] = True
                 sparse_contacts[j, i] = True
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef inline void _minimum_image_orthogonal(cython.floating[:] dx, cython.floating[:] box, cython.floating[:] inverse_box):
```

Comment:
Our current implementations in `calc_distances.h` are a weird mixed but mostly single precision.  I've rewritten these here using cython fused types (aka templates) to allow both single and double precision